virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -625,6 +634,41 @@ def allocate_gpp(self) -> None:
                 stem_traits=community.stem_traits, at_dbh=cohorts.dbh_values
             )
 
+    def apply_mortality(self) -> None:
+        """Apply mortality to plant cohorts.
+
+        This function applies the basic annual mortality rate to plant cohorts. The
+        mortality rate is currently a constant value for all cohorts. The function
+        calculates the number of individuals that have died in each cohort and updates
+        the cohort data accordingly. The function then updates deadwood production.
+
+        """
+
+        self.data["deadwood_production"] = xr.full_like(self.data["elevation"], 0)
+        self.data["deadwood_lignin"] = xr.full_like(self.data["elevation"], 0)
+
+        # Loop over each grid cell
+        for cell_id in self.communities.keys():
+            community = self.communities[cell_id]
+            cohorts = community.cohorts
+
+            # Calculate the number of individuals that have died in each cohort
+            mortality = np.round(
+                cohorts.n_individuals * self.per_update_interval_stem_mortality_rate
+            )
+
+            # Decrease size of cohorts based on mortality
+            cohorts.n_individuals = cohorts.n_individuals - mortality
+
+            # Update deadwood production
+            total_deadwood_mass = np.sum(mortality * community.stem_allometry.stem_mass)
+            self.data["deadwood_production"][cell_id] = total_deadwood_mass * (
+                1 - self.model_constants.percent_stem_mass_attributed_to_lignin
+            )
+            self.data["deadwood_lignin"][cell_id] = total_deadwood_mass * (
+                self.model_constants.percent_stem_mass_attributed_to_lignin
+            )
+
```

Comment:
As the lignin content is a proportion this calculation becomes more complex (i.e. it's the change in a proportion). I can send an equation over, but I guess it's worth asking whether that's worth implementing now or later, as at present we assume that all trees have the same lignin content anyway